### ISSUE TITLE
Japanese keyboard's spacebar description is larger than the other keyboard layouts

### ISSUE THREAD

**Issue by sudowesu** [florisboard/florisboard#2000]
While the other keyboard layouts' spacebar descriptions use a slight smaller and grey font like this,
![Screenshot_20220803-083241_Google.png](https://user-images.githubusercontent.com/67051709/182533126-626a28cf-8268-4300-838b-23b69d74a0b1.png)
日本語〔日本〕［JIS］keyboard uses a big white font in the spacebar in my device and therefore it doesn't fit.

#### Steps to reproduce
1. Install the 日本語〔日本〕［JIS］keyboard
2. See the error in the keyboard's spacebar

#### Environment information
- FlorisBoard Version: 0.3.16 (86)
- Install Source: GitHub
- Device: Samsung A10
- Android: 12, Custom ROM

**Comment by patrickgold** [comment]
Jup this seems to be the reason, thanks @GoRaN909 for pointing the issue out!

**Comment by patrickgold** [comment]
> You welcome bro i was replaced the japanese space code 12288 in my private forked FlorisBoard to code general space code 32 it worked like all other layouts too, Also that japanese space is different from the original one when you tap on it looks like three space insert ___ but the original is just one space _

The Japanese JIS layout has this space code for a reason, due to their symbols being wider than Latin symbols. Thus also their space is wider. I need to write a simple `isKeyCodeSpace()` fun to fix this issue.

**Comment by sudowesu** [comment]
Yes! Japanese space is full-width because every character is monospace. Actually, space key is mostly used for changing the characters to Kanji (more complicated ones), rather than spacing (because Japanese doesn't have space) but i think this would require different keyboards like GODAN, Japanese QWERTY, 12-keys etc. just like in Gboard. Do you have any plans on adding them? If so, should i create a different issue for it? I have already opened a few so i don't want to spam issues 😁

**Comment by RexPacificusVII** [comment]
![Image](https://github.com/user-attachments/assets/06109c22-b77d-4ffc-a195-99fa7acea916)

**Comment by RexPacificusVII** [comment]
![Image](https://github.com/user-attachments/assets/1b3d1931-307c-42d0-8c0e-25c890b29d2d)

If using a borderless theme, you can really see the label overlapping with other texts.

**Comment by klaurence** [comment]
I'm wondering why they are different. Even if the subtype language is Japanese, shouldn't it still display in the slightly smaller and grey font?

**Comment by lm41** [comment]
They are different because The monospaced Space and the normal Space are different `KeyCodes` and the Spacebar label code only checks for the normal space. This will be fixed with the new layouting. See: https://github.com/florisboard/florisboard/pull/2764#issuecomment-2698802056
